jdk/src/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -274,17 +294,55 @@ class Direct$Type$Buffer$RW$$BO$
             int rem = (pos <= lim ? lim - pos : 0);
             if (length > rem)
                 throw new BufferUnderflowException();
+            try {
+#if[!byte]
+                if (order() != ByteOrder.nativeOrder())
+                    Bits.copyTo$Memtype$Array(ix(pos), dst,
+                                          (long)offset << $LG_BYTES_PER_VALUE$,
+                                          (long)length << $LG_BYTES_PER_VALUE$);
+                else
+#end[!byte]
+                    Bits.copyToArray(ix(pos), dst, arrayBaseOffset,
```

Comment:
lost some code here

Reply:
Close - I duplicated some code here.  Not sure how I did it but I duplicated some of the methods